docs/contributor/adr/003-client-scope-and-usage.md

```diff
@@ -0,0 +1,109 @@
+
+Given this, we have choices:
+- Create our own simplified interfaces that align with our usage and ignore the embedded interfaces
+- Use the client.Client interface directly in the Repository layer
+- Create an internal interface that embeds a subset or all of the controller-runtime interfaces
+
+This decision will influence how we encapsulate data access logic aka kube API access and how flexible our system will be for testing.
+
+## Decision
+
+We will use the `client.Client` interface provided by `controller-runtime` directly in the Repository layer and avoid creating custom interfaces or client implementations. This approach minimizes unnecessary abstraction and leverages the mature, well-tested interfaces provided by the Kubernetes ecosystem.
+
+Whenever possible, we will reference the most specific sub-interface from the `client.Client` composition (e.g., `Reader`, `Writer`) rather than the full `Client` interface. This promotes better adherence to the Interface Segregation Principle and allows for more precise dependency injection, leading to simpler and more focused unit tests. If methods from multiple sub-interfaces are needed, the `client.Client` is used.
+
+The client will **only** be referenced in the Repository layer. All other layers, such as Service and Reconciler, will remain decoupled from infrastructure concerns and interact only through higher-level abstractions defined in the application domain.
+
+### Do's
+
+A Service defines its dependecy named *Repository. The Repository implementation then uses controller-runtime's Client interfaces directly.
+```go 
+type SomeService struct {
+	manifestRepository ManifestRepository
+}
+
+type ManifestRepository interface {
+	Get(name, namespace string) (*v1beta2.Manifest, error)
+}
+```
+```go
+type ManifestRepositoryImpl struct {
+	readClient client.Reader
+}
+```
+Another example, for more than only read:
+```go
+type ManifestRepository interface {
+    Get(name, namespace string) (*v1beta2.Manifest, error)
+    Create(name, namespace string) error
+    Update(*v1beta2.Manifest) error
+}
+```
+```go
+type ManifestRepositoryImpl struct {
+	kcpClient client.Client
+}
+```
+
+### Don'ts
+
+The Service defines a dependency named *Client and uses it:
+```go
+type SomeService struct {
+	manifestClient ManifestClient
+}
+
+type ManifestClient interface {
+	Get(name, namespace string) (*v1beta2.Manifest, error)
+}
+```
+Mitigation: It should be called *Repository.
+
+
+The Service consumes the defined ManifestRespository interface as an embedded field:
+```go
+type SomeService struct {
+	ManifestRepository
+}
+
+type ManifestRepository interface {
+	Get(name, namespace string) (*v1beta2.Manifest, error)
+}
+```
+
+Mitigation: It should be referenced as a named, private field so it can not be accessed across the package.
```

Comment:
```suggestion
   Mitigation: The `ManifestRepository` interface must be referenced as a named, private field so it can not be accessed across the package.
```